Onboarding Chunk 1

yougotwill opened the pull request.

yougotwill committed:
feat: upgrade react to v18

including eslint plugins to latest

yougotwill committed:
fix: use createRoot instead of ReactDom.render

yougotwill committed:
refactor: remove unnecessary react imports

you don't need it from react 17

yougotwill committed:
fix: pass yarn ready

fixed some weird import errors

yougotwill committed:
feat: add framer-motion

tested out styled components compatbility with the MessageHighlighter component

yougotwill committed:
refactor: clean up remaining global React import calls

import specific types from react package

yougotwill committed:
Merge pull request #39 from yougotwill/feat/ses-912/react18_and_framer_motion

Feat/ses 912/react18 and framer motion

yougotwill committed:
fix: clear unused animation on send button

yougotwill committed:
feat: moved spinners into new loading folder

converted session-loader sass to styled components

yougotwill committed:
feat: added SessionProgressBar

added copy for onboarding account loading

yougotwill committed:
feat: load SessionProgressBar component for testing

with already have an account copy

yougotwill committed:
feat: starting work on new input

added into test region of right panel

yougotwill committed:
feat: basic ui styling of new input

including improved error messaging and removing the label

yougotwill committed:
feat: fade in error colors

yougotwill committed:
fix: change frame duration to string

yougotwill committed:
feat: continue displaying input errors even if the user renters an input

yougotwill committed:
feat: migrate session inputs on registration screen to the new  component

still need to do the slide animation

yougotwill committed:
fix: show hide button position

yougotwill committed:
feat: added support for cta button

with sliding down animation if an error occurs

yougotwill committed:
feat: updated tos and privacy policy on registration link

intial run at legal links modal

yougotwill committed:
fix: remove TermsOfServicePrivacyDialog from convo state

we are moving it to a separate onboarding store

yougotwill committed:
feat: created onboarding store and added modals slice

yougotwill committed:
feat: improved TermsOfServicePrivacyDialog styling

added new modal confirm buttons, only tested in classic dark for onboarding

yougotwill committed:
feat: added correct links to modal

yougotwill committed:
feat: added responsive hero

cleaned up some sass and changed to styled components

yougotwill committed:
Merge branch 'unstable' into feat/ses-897/steps-create-account

yougotwill committed:
feat: updated hero image

we simplified the hero responsiveness

yougotwill committed:
feat: setup registration slice

yougotwill committed:
feat: replace context with redux slice

yougotwill committed:
feat: extract back button to separate component

yougotwill committed:
feat: rename onboarding stages for clarity

also comment on what changes need to be made within the different stages

yougotwill committed:
feat: updated onboarding header

added external link buttons, removed accentText and added new svgs

yougotwill committed:
feat: updated back button icon

yougotwill committed:
fix: doc string comments for registration slice

yougotwill committed:
feat: renamed stages components to match redux slices

setup stage, trying to fix back button position, moved out start logic from stages

yougotwill committed:
feat: fix back button positioning relative to related component

we can now pass children to a container that holds the back button

yougotwill committed:
feat: setup UI and state for account create display name step

yougotwill committed:
feat: updated display name

show error when display name is too long

yougotwill committed:
fix: rename MAX_USERNAME_BYTES to MAX_NAME_LENGTH to match libsession, updated byte limit to 100

added doctstring to explain the prop

yougotwill committed:
fix: replaced session input component with the new one

yougotwill committed:
refactor: moved related function from registrationstages to createaccount

yougotwill committed:
feat: make sure to set the account create state to done

yougotwill committed:
feat: refactored empty message view component

use styled components

yougotwill committed:
feat: start working on account create success screen

on conversations view

yougotwill committed:
feat: finished styling success screen

need to test logic a bit more to make sure that I am happy

yougotwill committed:
fix: empty convo sentence must be on 2 lines

yougotwill committed:
feat: update logic for showing the empty convo text

yougotwill committed:
fix: correct background color on empty message view

yougotwill committed:
feat: added svg fill color as well as clip and fill rules to session icon component

this lets us render more complex svgs like the recovery password icon

yougotwill committed:
feat: move svg props to the icons file

Audric these are the files to cherry pick

yougotwill committed:
feat: updated recovery password bannner

yougotwill committed:
feat: styled start conversation button

yougotwill committed:
fix: classic light button should go to black on hover

yougotwill committed:
feat: updated restore account screen with new design

yougotwill committed:
fix: styled password show hide correctly

yougotwill committed:
feat: added error messages to recovery password input

fixed remaining styling, improved errors for mnemonic code

yougotwill committed:
feat: added both recovery password icon types

yougotwill committed:
feat: added in loading animation when restoring an account

yougotwill committed:
feat: animated progress loader to spec

timeout display name fetch to 15 seconds, added display name inital screen still needs work

yougotwill committed:
feat: display name screen shows correctly now

hide back button while loading

yougotwill committed:
feat: moved large useEffect to it's own hook

useRecoveryProgressEffect

yougotwill committed:
feat: dont show new account success when restoring

moved useRecoveryProgressEffect to a hooks folder

yougotwill committed:
feat: fixed alignment of loading screen and updated enter new display name copy

yougotwill committed:
fix: error handling now works correctly for all stages

buttons also correctly disable themselves

yougotwill committed:
fix: disable inputs when button is disabled

yougotwill committed:
feat: confirmed sign up with new display name works

yougotwill committed:
Merge branch 'unstable' into feat/ses-50/onboarding

yougotwill committed:
feat: started initial work on onboarding step animation

yougotwill committed:
feat: intial animation working correctly

yougotwill committed:
feat: finished fade animation between steps

moved useState props to redux, fixed back button

yougotwill committed:
Merge branch 'unstable' into feat/ses-50/onboarding

yougotwill committed:
feat: attempting to fetch the display name separate from the usual pipeline

yougotwill committed:
feat: added free function to libsession wrappers

almost processing the dump correctly

yougotwill committed:
feat: fixed processing the config message next is finishing the reg properly

we now fetch the display name correctly just need to refactor and test the error handling

yougotwill committed:
feat: got things working correctly with errors throwing before the timeout

still trying to debug loading the errors in the frontend

yougotwill committed:
feat: moved frontend errors to redux

yougotwill committed:
fix: add missing dispatch for set recovery pw error

yougotwill committed:
feat: remove pubkey text scrambler

since we no longer use it

yougotwill committed:
feat: sort messages json keys to make merging with strings project easier in future

yougotwill committed:
feat: session id to account id

yougotwill committed:
feat: recovery phrase to recovery password

yougotwill committed:
feat: continue your session and link device to I have an account

yougotwill committed:
feat: password to lock app

yougotwill committed:
feat: start page and tos ids updated

yougotwill committed:
feat: new account ids added

yougotwill committed:
feat: restore account ids added

yougotwill committed:
Merge branch 'feat/ses-50/onboarding' into feat/ses-899/user_profile_poll

yougotwill committed:
feat: replace invalidAccountId with accountIdErrorInvalid

as based in google docs flagged sheet

yougotwill committed:
fix: small fixes from broken branch as we try to debug

yougotwill committed:
feat: depracate getLastProfileUpdateTimestamp

since new user config has been on for a few months

yougotwill committed:
feat: reset onboarding state when we return to the start stage

added better logging for create account flow

yougotwill committed:
feat: working on restoring accounts with fetch

loads correctly but isn't saved to db and is lost on refresh

yougotwill committed:
fix: got correct order for restoring an account with fetch

yougotwill committed:
feat: restoring an account from seed and fetching seems to work

yougotwill committed:
feat: cleaned up poll once for display name

yougotwill committed:
fix: use wrapper merge when returning the display name in memory

yougotwill committed:
Merge branch 'unstable' into feat/ses-899/user_profile_poll

yougotwill committed:
feat: handle failing to find a display name from the swarm

the user can enter it manually, improved error handling, logging, sync post login

yougotwill committed:
fix: cleaned up typings and flattened error callback

yougotwill committed:
fix: dont fade animation between loading states

yougotwill committed:
fix: fixed animation for different steps

flatten the loading callback as well

yougotwill committed:
fix: hero image is now correctly responsive for both width and height

yougotwill committed:
fix: cleanup

yougotwill committed:
feat: refactored swarm polling to use only retrieveNextMessages again

created verifyBatchRequestResults function

yougotwill committed:
fix: cleanup

yougotwill committed:
feat: setup what tests need to be done with TODOs

yougotwill committed:
fix: moved resetRegistration out of utils to make unit testing easier

yougotwill committed:
fix: rename Registration_test to Onboarding_test

yougotwill committed:
fix: lint fixes

yougotwill committed:
Merge branch 'unstable' into feat/ses-1560/unit-tests

yougotwill committed:
fix: padding resoluton for onboarding buttons

yougotwill committed:
fix: alignment of ShowHideButton

yougotwill committed:
feat: added stubbing for window.i18n

yougotwill committed:
test: displayNameIsValid

yougotwill committed:
test: sanitizeSessionUsername

which is the core function of sanitizeDisplayNameOrToast in onboarding

yougotwill committed:
test: registerSingleDevice

started on signInByLinkingDevice

yougotwill committed:
test: validity check for signInByLinkedDevice done

made stubs for window whisper and storage

yougotwill committed:
test: finished signInByLinkingDevice

yougotwill committed:
fix: remove registration test

we dont do redux based unit testing at this time

yougotwill committed:
fix: remove pollOnceForOurDisplayName test

saved work in a branch but we need to use workers in unit tests before its work testing

yougotwill committed:
Merge branch 'unstable' into feat/ses-50/onboarding

Bilb commented on ts/components/registration/hooks/index.tsx:
```diff
@@ -0,0 +1,92 @@
+  }, [displayName, ourPubkey]);
+
+  useEffect(() => {
+    if (step === AccountRestoration.Loading) {
+      interval = setInterval(() => {
+        if (progress < totalProgress) {
+          dispatch(setProgress(progress + 1));
+        }
+
+        if (progress >= totalProgress) {
+          clearInterval(interval);
+          // if we didn't get the display name in time, we need to enter it manually
+          dispatch(setAccountRestorationStep(AccountRestoration.DisplayName));
+        }
+      }, ONBOARDING_TIMES.RECOVERY_TIMEOUT / totalProgress);
```
something like
```ts
export const useRecoveryProgressEffect = (props: UseRecoveryProgressEffectProps) => {
  const { step, progress, setProgress, ourPubkey, displayName } = props;
  const totalProgress = 100;

  const dispatch = useDispatch();

  const recoveryComplete = useCallback(() => {
    void finishRestore(ourPubkey, displayName);
  }, [displayName, ourPubkey]);

  const tickEvery = step === AccountRestoration.Loading ? ONBOARDING_TIMES.RECOVERY_TIMEOUT / totalProgress : step === AccountRestoration.Finishing ?ONBOARDING_TIMES.RECOVERY_FINISHING / totalProgress : step === AccountRestoration.Finished ? ONBOARDING_TIMES.RECOVERY_FINISHED : 0

  useInterval(() => {
    switch (step) {
      case AccountRestoration.Loading:
        if (progress < totalProgress) {
          dispatch(setProgress(progress + 1));
        }

        if (progress >= totalProgress) {
          // if we didn't get the display name in time, we need to enter it manually
          dispatch(setAccountRestorationStep(AccountRestoration.DisplayName));
        }
        break;
      case AccountRestoration.Finishing:
        if (progress < totalProgress) {
          dispatch(setProgress(progress + 1));
        }

        if (progress >= totalProgress) {
          dispatch(setAccountRestorationStep(AccountRestoration.Finished));
        }
        break;
      case AccountRestoration.Finished:
        if (!isEmpty(displayName)) {
          dispatch(setAccountRestorationStep(AccountRestoration.Complete));
        } else {
          // if we didn't get the display name in time, we need to enter it manually
          dispatch(setAccountRestorationStep(AccountRestoration.DisplayName));
        }
        break;
      case AccountRestoration.Complete:
        if (!isEmpty(ourPubkey) && !isEmpty(displayName)) {
          void recoveryComplete();
        } else {
          window.log.debug(`[onboarding] restore account: We don't have a pubkey or display name`);
          dispatch(setAccountRestorationStep(AccountRestoration.DisplayName));
        }
        break;
      default:
        break;
    }
  }, tickEvery)

};
```

Bilb commented on ts/components/registration/stages/CreateAccount.tsx:
```diff
@@ -0,0 +1,157 @@
+
+    try {
+      window.log.debug(
+        `WIP: [onboarding] create account: signUp() is starting display name: ${displayName} recoveryPassword: ${recoveryPassword}`
```
probably some log cleaning needed, maybe a chunk 2

Bilb commented on ts/components/registration/stages/CreateAccount.tsx:
```diff
@@ -0,0 +1,157 @@
+          type="text"
+          placeholder={window.i18n('enterDisplayName')}
+          value={displayName}
+          onValueChanged={(_name: string) => {
+            const name = sanitizeDisplayNameOrToast(_name, setDisplayNameError, dispatch);
+            dispatch(setDisplayName(name));
+          }}
```
usually I use _vars for unused ones
```suggestion
          onValueChanged={(name: string) => {
            const satinizedName = sanitizeDisplayNameOrToast(name, setDisplayNameError, dispatch);
            dispatch(setDisplayName(satinizedName));
          }}
```

Bilb commented on ts/components/registration/stages/CreateAccount.tsx:
```diff
@@ -0,0 +1,157 @@
+  });
+
+  const signUpWithDetails = async () => {
+    if (!(!!displayName && !displayNameError)) {
```
you check for that same exact condition  on line 151. You can probably remove it from down there, but keep this one.

Also, this is equivalent to 
!displayName || displayNameError
right?

Bilb replied:
What I meant is that the whole component returns early if 
`!(!!displayName && !displayNameError)` is true

So the disabled state is never reached below (as we return early when that condition is true!)
You need to keep the one returning early, but not the one on the disabled state

yougotwill replied:
There is no early return in the component `isEmpty(displayName) || !isEmpty(displayNameError)` happens in `signUpWithDetails` and the button disabled state so we need both. Resolved

Bilb commented on ts/components/registration/stages/RestoreAccount.tsx:
```diff
@@ -0,0 +1,321 @@
+ * Ask for a display name, as we will drop incoming ConfigurationMessages if any are saved on the swarm.
+ * We will handle a ConfigurationMessage
+ */
+async function signInWithNewDisplayName(args: AccountRestoreDetails) {
+  const { displayName, recoveryPassword, dispatch } = args;
```
is this a new way of destructuring?
What happened to the 
```suggestion
async function signInWithNewDisplayName({ displayName, recoveryPassword, dispatch } : AccountRestoreDetails) {

```
syntax? We don't like it anymore?

Bilb commented on ts/components/registration/stages/RestoreAccount.tsx:
```diff
@@ -0,0 +1,321 @@
+}
+
+export const RestoreAccount = () => {
+  const step = useOnboardAccountRestorationStep();
+  const recoveryPassword = useRecoveryPassword();
+  const recoveryPasswordError = useRecoveryPasswordError();
+  const ourPubkey = useOnboardHexGeneratedPubKey();
+  const displayName = useDisplayName();
+  const displayNameError = useDisplayNameError();
+  const progress = useProgress();
+
+  const dispatch = useDispatch();
+
+  useRecoveryProgressEffect({
+    step,
+    progress,
+    setProgress,
+    ourPubkey,
+    displayName,
+  });
```
you could move some of those hooks call in the `useRecoveryProgressEffect` and return them from the hook rather than passing them. It's a bit weird to pass something to the hook which is changed by the hook.

So something like

```ts

const {step, progress} =  useRecoveryProgressEffect({
//    setProgress, it's just an import that you can do in the hook itself too
    ourPubkey,
    displayName,
  });

Bilb commented on ts/components/registration/stages/RestoreAccount.tsx:
```diff
@@ -0,0 +1,321 @@
+                  type="text"
+                  placeholder={window.i18n('enterDisplayName')}
+                  value={displayName}
+                  onValueChanged={(_name: string) => {
+                    const name = sanitizeDisplayNameOrToast(_name, setDisplayNameError, dispatch);
+                    dispatch(setDisplayName(name));
```
same comment, use something else than the _ syntax for variables used

Bilb commented on ts/components/registration/stages/RestoreAccount.tsx:
```diff
@@ -0,0 +1,321 @@
+                  buttonColor={SessionButtonColor.White}
+                  onClick={recoverAndEnterDisplayName}
+                  text={window.i18n('continue')}
+                  disabled={
+                    !(!!recoveryPassword && !recoveryPasswordError) ||
+                    !(!!displayName && !displayNameError)
+                  }
```
this is pretty hard to understand.
it's not the equivalent of 
```suggestion
                  disabled={
                    !recoveryPassword || recoveryPasswordError || !displayName || displayNameError
                  }
```

Bilb replied:
yep that's better

Bilb commented on ts/session/apis/snode_api/swarmPolling.ts:
```diff
@@ -617,4 +676,71 @@ export class SwarmPolling {
+      }
+
+      const pubkey = UserUtils.getOurPubKeyFromCache();
+      const toPollFrom = await this.getNodesToPollFrom(pubkey.key);
+
+      if (abortSignal?.aborted) {
+        throw new NotFoundError(
+          '[pollOnceForOurDisplayName] aborted after selecting nodes to poll from'
+        );
+      }
+
+      const resultsFromUserProfile = await SnodeAPIRetrieve.retrieveNextMessages(
+        toPollFrom,
+        [''],
+        pubkey.key,
+        [SnodeNamespaces.UserProfile],
+        pubkey.key,
+        null
+      );
```
we only want to poll from the namespace of the user profile right?
this.getNodesToPollFrom will return all the user's namespaces, including the 0 one.

Bilb commented on ts/session/apis/snode_api/swarmPolling.ts:
```diff
@@ -400,6 +404,44 @@ export class SwarmPolling {
           `handleConfigMessagesViaLibSession of "${allDecryptedConfigMessages.length}" messages with libsession`
         );
+
+        if (returnDisplayNameOnly) {
+          let displayName = '';
+
+          try {
+            const keypair = await UserUtils.getUserED25519KeyPairBytes();
+            if (!keypair || !keypair.privKeyBytes) {
+              throw new Error('edkeypair not found for current user');
+            }
+
+            const privateKeyEd25519 = keypair.privKeyBytes;
+
+            // we take the lastest config message to create the wrapper in memory
+            const incomingConfigMessages = allDecryptedConfigMessages.map(m => ({
+              data: m.message.data,
+              hash: m.messageHash,
+            }));
+
+            await GenericWrapperActions.init('UserConfig', privateKeyEd25519, null);
+            await GenericWrapperActions.merge('UserConfig', incomingConfigMessages);
+
+            const userInfo = await UserConfigWrapperActions.getUserInfo();
+            if (!userInfo) {
+              throw new Error('UserInfo not found');
+            }
+            displayName = userInfo.name;
+          } catch (e) {
+            window.log.warn(
+              'LibSessionUtil.initializeLibSessionUtilWrappers failed with',
+              e.message
+            );
+          } finally {
+            await GenericWrapperActions.free('UserConfig');
+          }
+
+          return displayName;
+        }
```
You can even do
```suggestion
        if (returnDisplayNameOnly) {

          try {
            const keypair = await UserUtils.getUserED25519KeyPairBytes();
            if (!keypair || !keypair.privKeyBytes) {
              throw new Error('edkeypair not found for current user');
            }

            const privateKeyEd25519 = keypair.privKeyBytes;

            // we take the lastest config message to create the wrapper in memory
            const incomingConfigMessages = allDecryptedConfigMessages.map(m => ({
              data: m.message.data,
              hash: m.messageHash,
            }));

            await GenericWrapperActions.init('UserConfig', privateKeyEd25519, null);
            await GenericWrapperActions.merge('UserConfig', incomingConfigMessages);

            const userInfo = await UserConfigWrapperActions.getUserInfo();
            if (!userInfo) {
              throw new Error('UserInfo not found');
            }
            return userInfo.name;
          } catch (e) {
            window.log.warn(
              'LibSessionUtil.initializeLibSessionUtilWrappers failed with',
              e.message
            );
          } finally {
            await GenericWrapperActions.free('UserConfig');
          }
          return '';
        }
```

and I think the finally would still be called and the return in the try be returned :exploding_head:

Bilb commented on ts/session/apis/snode_api/swarmPolling.ts:
```diff
@@ -617,4 +676,71 @@ export class SwarmPolling {
+
+      return displayName;
+    } catch (e) {
+      if (e.message === ERROR_CODE_NO_CONNECT) {
+        if (window.inboxStore?.getState().onionPaths.isOnline) {
+          window.inboxStore?.dispatch(updateIsOnline(false));
+        }
+      } else if (!window.inboxStore?.getState().onionPaths.isOnline) {
+        window.inboxStore?.dispatch(updateIsOnline(true));
+      }
```
we don't need this part when we just poll for our display name (the bottom left status light is not displayed, and that inboxStore is actually not setup)

Bilb commented on ts/components/icon/Icons.tsx:
```diff
@@ -403,6 +444,11 @@ export const icons: Record<string, { path: string; viewBox: string; ratio: numbe
     ratio: 1,
   },
+  question: {
+    path: 'M4.70711 5.34466C5.12711 5.34466 5.47244 5.19066 5.74311 4.88266C6.02311 4.57466 6.16311 4.16866 6.16311 3.66466V3.58066C6.16311 3.30066 6.11178 3.03932 6.00911 2.79666C5.91578 2.55399 5.78044 2.34399 5.60311 2.16666C5.42578 1.97999 5.21111 1.83532 4.95911 1.73266C4.70711 1.62066 4.42711 1.56466 4.11911 1.56466C3.44711 1.56466 2.91978 1.76999 2.53711 2.18066C2.15444 2.59132 1.96311 3.1 ...
+    viewBox: '0 0 8 11',
+    ratio: 1,
+  },
```
preview was probably just a svgomg comparison.

yougotwill committed:
fix: renamed button modal-confirm to ghost

moved SessionButton defaults into component

yougotwill committed:
fix: simplified SpacerStyled component

added matching css variable to each spacers docstring

yougotwill committed:
fix: backbuttonwithinincontainer to backbuttonwithincontainer

yougotwill committed:
fix: make useRecoveryProgressEffect use a switch for better readability

yougotwill committed:
fix: used better naming for value changed in create account

cleaned up logging

yougotwill committed:
fix: destructure args in sign in methods

cleaned up logging

yougotwill committed:
fix: dont pass redux args to restoration hook

we can use the selectors inside of it

yougotwill committed:
fix: used better naming for value changed in restore account

yougotwill committed:
fix: improved check for disabling continue button

more human readable

yougotwill committed:
fix: turn off enableLogRedirect

yougotwill committed:
fix: updated pollOnceForDisplayName

we can return the try result directly in the block since the finally will alway be executed

yougotwill committed:
fix: remove unneeded catch block for checking if we are online

the inboxStore isnt set yet and we arent displaying the connected path light

yougotwill committed:
feat: converted stagedGenericAttachment to be functional

yougotwill committed:
chore: bump libsession_util_nodejs to v0.3.3

typings for free() on a wrapper which we use to clear in memory only wrappers in some cases

yougotwill committed:
fix: moved some things outside of the OnboardContainer

this prevents refreshing of the entire styled component on render

yougotwill committed:
fix: compressed all new svgs

tweaked svg styling

yougotwill committed:
fix: repaired rounding on upload profile image component

yougotwill committed:
fix: prevent flickering on EmptyMessageView

yougotwill committed:
fix: hr in emptymessageview should not use 0.5px

yougotwill committed:
Merge branch 'unstable' into feat/ses-50/onboarding

yougotwill committed:
fix: right panel transition is now smoother

transition duration is now set via a function for integration testing  purposes

yougotwill committed:
chore: bump libsession_util_nodejs to v0.3.4

fixes builds on windows

Bilb commented on ts/components/basic/Text.tsx:
```diff
@@ -26,62 +26,84 @@ export const Text = (props: TextProps) => {
+              ? 'var(--margins-md)'
+              : props.size === 'sm'
+                ? 'var(--margins-sm)'
+                : 'var(--margins-xs)'};
```
works for me

Bilb commented on ts/components/basic/SessionButton.tsx:
```diff
@@ -1,16 +1,18 @@
   Outline = 'outline',
   Simple = 'simple',
   Solid = 'solid',
+  ModalConfirm = 'modal-confirm',
```
nice

Bilb commented on ts/components/registration/RegistrationStages.tsx:
```diff
@@ -26,222 +28,76 @@ export async function resetRegistration() {
+              style={{ border: '2px solid var(--text-primary-color)', borderRadius: '9999px' }}
+              dataTestId="session-faq-link"
+              onClick={() => {
+                void shell.openExternal('https://getsession.org/faq');
```
ok

Bilb commented on ts/components/registration/hooks/index.tsx:
```diff
@@ -0,0 +1,92 @@
+} from '../../../state/onboarding/ducks/registration';
+import { finishRestore } from '../stages/RestoreAccount';
+
+let interval: NodeJS.Timeout;
```
that's alright. Don't worry about it.

Bilb commented on ts/components/loading/bar/SessionProgressBar.tsx:
```diff
@@ -0,0 +1,84 @@
+`;
+
+const StyledTitle = styled(StyledText)`
+  font-size: 26px;
```
I just meant that I'd expect font-size to be var(--something)

yougotwill replied:
We don't have a size 26px font size or anything close to that and that font size is only used in one place (as you said above it's custom. Are you happy for me to leave it?

yougotwill replied:
resolving

Bilb commented on ts/components/EmptyMessageView.tsx:
```diff
@@ -0,0 +1,124 @@
+`;
+
+export const EmptyMessageView = () => {
+  const conversationCount = useSelector(getLeftPaneConversationIdsCount);
```
all good, there is no `useLeftPaneConversationIdsCount ` so we can keep it  this way.

Bilb commented on ts/components/inputs/SessionInput.tsx:
```diff
@@ -0,0 +1,231 @@
+        onClick={props.toggleForceShow}
+        style={{
+          marginTop: '-0.5px',
+          marginRight: '0.25px',
```
I got those from my good personnal friend, could you give them a go and see if that fixes it?
![eyeBlinded](https://github.com/oxen-io/session-desktop/assets/1544279/6d4533b6-5de3-4ccc-9598-f761d54b129a)
![eye](https://github.com/oxen-io/session-desktop/assets/1544279/542aa00d-e871-4f7a-a067-6783c4d4eea4)

yougotwill replied:
Fixed!

yougotwill committed:
fix: replaced eye svgs with better aligned ones

fixed style position

yougotwill committed:
fix: sessioninput color should be danger when there is an error

brought improvements from chunk 2

yougotwill committed:
fix: use recoveryPasswordOutline for light themes

updated to use building block svgs

yougotwill committed:
fix: match messageNewDescription copy to figma

even though its chunk 2

yougotwill committed:
fix: replace setAccountPasswordTitle with lockApp string

yougotwill committed:
fix: use correct hero image

yougotwill committed:
fix: welcome message should use text-primary-color for light themes

yougotwill committed:
fix: repair placeholder selectors

required since upgrading styled-components

yougotwill committed:
fix: brand logo in onboard header should not block buttons

yougotwill committed:
fix: ses-2087 recovery password banner heading is too large

yougotwill committed:
fix: ses-2089 update tos datatestid to open-url to match other platforms

yougotwill committed:
feat: updated hero image

yougotwill committed:
Merge branch 'unstable' into feat/ses-50/onboarding

improved global window types for persistStore and inboxStore, we now await the persist store purge in deleteDbLocally

yougotwill committed:
fix: revert swarmpolling to before merging unstable

Bilb commented:
lgtm
